aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -100,22 +98,31 @@ public static ModelProviderStep builder() {
      */
     Completable hydrate() {
         final List<Completable> hydrationTasks = new ArrayList<>();
-        List<ModelSchema> modelSchemas = new ArrayList<>(modelProvider.modelSchemas().values());
+        List<ModelSchema> modelSchemas = Observable.fromIterable(modelProvider.modelSchemas().values())
+            // Don't sync models that have an associated syncExpression of QueryPredicates.none().
+            .filter(schema -> !QueryPredicates.none().equals(queryPredicateProvider.getPredicate(schema.getName())))
```

Comment:
instead of doing a direct comparison on `QueryPredicates.none()`, maybe it's better to use `x instanceof MatchNoneQueryPredicate` instead? 

There is no guarantee that `QueryPredicates.none()` is a constant (even though it is) since it's a static method result right? Whereas `MatchNoneQueryPredicate.class` is a constant.

Reply:
Even if it's not a constant, it should always return the correct result, because of the [implementation of `equals`](https://github.com/aws-amplify/amplify-android/blob/b29c39ee9b73f359297f93567541e11a81812fc1/core/src/main/java/com/amplifyframework/core/model/query/predicate/MatchNoneQueryPredicate.java#L48), which returns true as long as `obj instanceof MatchNoneQueryPredicate`.

I did a quick search, and it looks like we currently use the `QueryPredicates.all().equals(queryPredicate)` approach in 8 places, and the `queryPredicate instanceof MatchAllQueryPredicate` approach in 3 places.  I actually think we should make the `MatchAllQueryPredicate` and `MatchNoneQueryPredicate` classes package private.  I think they are implementation details, and we should only access them via the `QueryPredicates` factory.

Reply:
Here's a separate PR that makes those classes package-private: https://github.com/aws-amplify/amplify-android/pull/1127